### ISSUE TITLE
API endpoint for HKU LDSHE

### ISSUE THREAD

**Issue by ctchanandy** [nadavkav/moodle-local_learningdesigner_import#1]
Hello @nadavkav, this is Andy from HKU CITE. Sorry for the late response on this wonderful Moodle plugin. I think you might already found a way to retrieve the JSON representation of a "LD" in LDSHE. 

Actually we also have an API endpoint to retrieve it after login by this URL https://ldshe.cite.hku.hk/api/lds/designs/[ID]. We did have very primitive OAuth built-in but it is a bit outdated, but this endpoint is at least an easier way to get a clean JSON file.

**Comment by nadavkav** [comment]
Hello @ctchanandy , thank for the information! it is very helpful. the endpoint works great.

I started exploring the JSON schema with https://jsoneditoronline.org/ to figure out what would be a basic starting point for me to be able to import a generic course design I made in LDSHE into a Moodle course.

When we finish an initial phase of making it all work with the open endpoint, it will make better sense to add some basic or more advanced authentication. of course. but for now, it's great!
